includes/class-wc-payments-order-service.php

```diff
@@ -1657,12 +1657,13 @@ private function generate_dispute_created_note( $charge_id, $amount, $reason, $d
 				/* translators: %1: the disputed amount and currency; %2: the dispute reason; %3 the deadline date for responding to dispute */
 				__( 'Payment has been disputed for %1$s with reason "%2$s". <a>Response due by %3$s</a>.', 'woocommerce-payments' ),
 				[
-					'a' => '<a href="' . $dispute_url . '" target="_blank" rel="noopener noreferrer">',
+					'a' => '<a href="%4$s" target="_blank" rel="noopener noreferrer">',
```

Comment:
This is a nice refactor! It may have missed my eye since it is within an array value, but I will keep an eye out if I write or come across similar code!